src/dataflow/mod.rs

```diff
@@ -2,135 +2,48 @@
 //!
 //! For a user-centric version of how to define a dataflow, read the
 //! `bytewax.dataflow` Python module docstring. Read that first.
+//!
+//! This is a "blueprint" of a dataflow. We compile this into a Timely
+//! dataflow in [`crate::execution::build_production_dataflow`]. We
+//! can't call into this structure directly from Timely because PyO3
+//! does not like generics.
 
 use crate::inputs::InputConfig;
 use crate::outputs::OutputConfig;
 use crate::pyo3_extensions::TdPyCallable;
+use crate::recovery::StepId;
 use crate::window::ClockConfig;
 use crate::window::WindowConfig;
 use pyo3::exceptions::PyValueError;
 use pyo3::prelude::*;
 use pyo3::types::*;
-use serde::Deserialize;
-use serde::Serialize;
-use sqlx::encode::IsNull;
-use sqlx::error::BoxDynError;
-use sqlx::sqlite::SqliteArgumentValue;
-use sqlx::sqlite::SqliteTypeInfo;
-use sqlx::sqlite::SqliteValueRef;
-use sqlx::Decode;
-use sqlx::Encode;
-use sqlx::Sqlite;
-use sqlx::Type;
-use std::borrow::Cow;
-use std::fmt::Display;
-
-/// Newtype representing the unique name of a step in a dataflow.
-///
-/// Used as a key for looking up relevant state for different steps
-/// surrounding recovery.
-#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize)]
-pub(crate) struct StepId(String);
```

Comment:
Moved into `crate::recovery` since the reason it exists is to facilitate recovery. And then all the SQLx stuff can be isolated in there.